docs/concepts/overview/extending.md

```diff
@@ -0,0 +1,217 @@
+
+The remainder of this guide is about Extensions.
+
+## Extensions
+
+Extensions are software components that extend and deeply integrate with kubernetes. 
+They adapt it to support new types and new kinds of hardware.
+
+Most cluster administrators will use a hosted or distribution 
+instance of Kubernetes.  As a result, most Kubernetes users will need to 
+install extensions and fewer will need to author new ones.
+
+## Extension Patterns
+
+Kubernetes is designed to be automated by writing client programs.  Any
+program that reads and/or writes to the Kubernetes API can provide useful
+automation.  **Automation** can run on the cluster or off it.  By following
+the guidance in this doc you can write highly available and robust automation.
+Automation generally works with any Kubernetes cluster, including hosted
+clusters, and managed installations.
+
+There is a specific pattern for writing client programs that work well with
+Kubernetes called the **Controller** pattern.  Controllers typically read an
+object's `.spec`, possibly do things, and then update the object's `.status`.
+
+A controller is a client of Kubernetes.  When Kubernetes is the client and
+calls out to a remote service, it is called a **Webhook.  **The remote service
+is called a **Webhook Backend**.  Like Controllers, Webhooks do add a point of
+failure.  
+
+In the webhook model, Kubernetes makes a network request to a remote service.
+In the **Binary Plugin** model, kubernetes executes a binary (program).
+Binary plugins are used by the kubelet (e.g. [Flex Volume
+Plugins](https://github.com/kubernetes/community/blob/master/contributors/devel/flexvolume.md)
+and [Network
+Plugins](/docs/concepts/cluster-administration/network-plugins/))
+and by kubectl.  Plugin interfaces are still evolving, and may change.
+
+Below is a diagram showing how the extensions points interact with the
+Kubernetes control plane.
+
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vQBRWyXLVUlQPlp7BvxvV9S1mxyXSM6rAc_cbLANvKlu6kCCf-kGTporTMIeG5GZtUdxXz1xowN7RmL/pub?w=960&h=720">
+
+<!-- image source drawing https://docs.google.com/drawings/d/1muJ7Oxuj_7Gtv7HV9-2zJbOnkQJnjxq-v1ym_kZfB-4/edit?ts=5a01e054 -->
+
+
+## Extension Points
+
+This diagram shows the extension points in a Kubernetes system.  
+
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vSH5ZWUO2jH9f34YHenhnCd14baEb4vT-pzfxeFC7NzdNqRDgdz4DDAVqArtH4onOGqh0bhwMX0zGBb/pub?w=425&h=809">
+
+<!-- image source diagrams: https://docs.google.com/drawings/d/1k2YdJgNTtNfW7_A8moIIkij-DmVgEhNrn3y2OODwqQQ/view -->
+
+*   Kubectl plugins extend the kubectl binary ①.  [Link for 1.8 docs branch: docs/tasks/extend-kubectl/kubectl-plugins ] They only affect the individual user's local environment.  These extension points are described in the [Client Extensions](todo) section.
```

Comment:
**Make sure all of the links are no longer "todo" and are correct.**

Also, use an ordered list (numbers instead of bullet points) to match the numbers in the diagram, i.e.:

```
1. Kubectl plugins extend the kubectl binary.  See [Extend kubectl with plugins](/docs/tasks/extend-kubectl/kubectl-plugins/) They only affect the individual user's local environment.  These extension points are described in the [Client Extensions](todo) section.
1. The apiserver handles all requests.  Several types of extension points in the apiserver allow authenticating requests, or blocking them based on their content, editing content, and handling deletion.  These are described in the [API Access Extensions](todo) section.
1. ...
```

Which will render as:

1. Kubectl plugins extend the kubectl binary.  See [Extend kubectl with plugins](/docs/tasks/extend-kubectl/kubectl-plugins/). They only affect the individual user's local environment.  These extension points are described in the [Client Extensions](todo) section.
1. The apiserver handles all requests.  Several types of extension points in the apiserver allow authenticating requests, or blocking them based on their content, editing content, and handling deletion.  These are described in the [API Access Extensions](todo) section.
1. ...

Reply:
To @chenopis's excellent review I would add:
- Include only one space after periods.
- Glossary additions are a great idea!